src/python/pants/backend/docker/tailor_test.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.backend.docker.tailor import PutativeDockerTargetsRequest
+from pants.backend.docker.tailor import rules as docker_tailor_rules
+from pants.backend.docker.target_types import DockerImage
+from pants.core.goals.tailor import (
+    AllOwnedSources,
+    PutativeTarget,
+    PutativeTargets,
+    PutativeTargetsSearchPaths,
+)
+from pants.engine.rules import QueryRule
+from pants.testutil.rule_runner import RuleRunner
+
+
+def test_find_putative_targets() -> None:
+    rule_runner = RuleRunner(
+        rules=[
+            *docker_tailor_rules(),
+            QueryRule(PutativeTargets, [PutativeDockerTargetsRequest, AllOwnedSources]),
+        ],
+        target_types=[
+            DockerImage,
+        ],
+    )
+    rule_runner.create_file("src/docker_ok/Dockerfile")
+    rule_runner.create_file("src/docker_orphan/Dockerfile")
+    pts = rule_runner.request(
+        PutativeTargets,
+        [
+            PutativeDockerTargetsRequest(PutativeTargetsSearchPaths(("src/",))),
+            AllOwnedSources(
+                [
+                    "src/docker_ok/Dockerfile",
+                ]
```

Comment:
FYI: Black is formatting this weirdly because of the trailing comma, which it takes as a cue from you that you want things on a new line. If you remove the trailing comma, this will be formatted into a single line.

It's not wrong to have this and this is not blocking, but fixing it (and a few other places in this file) may make things more succinct.